ext/src/Map.c

```diff
@@ -518,11 +518,14 @@ php_driver_map_compare(zval *obj1, zval *obj2 TSRMLS_DC)
   }
 
   HASH_ITER(hh, map1->entries, curr, temp) {
-    php_driver_map_entry *entry;
+    php_driver_map_entry *entry = NULL;
     HASH_FIND_ZVAL(map2->entries, PHP5TO7_ZVAL_MAYBE_P(curr->key), entry);
     if (entry == NULL) {
       return 1;
     }
```

Comment:
Yeah, I thought about this when originally implementing it and favored the fast and easy to implement route. I think that documenting the limitation might be a better route here. What do you think?